src/fragmentarium/ui/fragment/WordExport.tsx

```diff
@@ -0,0 +1,428 @@
+import React from 'react'
+
+import { Fragment } from 'fragmentarium/domain/fragment'
+import Record from 'fragmentarium/ui/info/Record'
+import {
+  Document,
+  HeadingLevel,
+  Paragraph,
+  TextRun,
+  TableCell,
+  TableRow,
+  Table,
+  BorderStyle,
+  WidthType,
+  AlignmentType,
+  HyperlinkRef,
+  HyperlinkType,
+  FootnoteReferenceRun,
+} from 'docx'
+import { ReactElement } from 'react'
+import TransliterationLines from 'transliteration/ui/TransliterationLines'
+import TransliterationNotes from 'transliteration/ui/TransliterationNotes'
+import { Glossary } from 'transliteration/ui/Glossary'
+import { renderToString } from 'react-dom/server'
+import $ from 'jquery'
+import rgbHex from 'rgb-hex'
+import WordService from 'dictionary/application/WordService'
+import GlossaryFactory from 'transliteration/application/GlossaryFactory'
+import { MemoryRouter } from 'react-router-dom'
+
+export async function wordExport(
+  fragment: Fragment,
+  wordService: WordService
+): Promise<Document> {
+  const tableHtml: JQuery = $(
+    renderToString(TransliterationLines({ text: fragment.text }))
+  )
+  const notesHtml: JQuery = $(
+    renderToString(TransliterationNotes({ notes: fragment.text.notes }))
+  )
+  const records: JQuery = $(
+    renderToString(Record({ record: fragment.uniqueRecord }))
+  )
+
+  const glossaryFactory: GlossaryFactory = new GlossaryFactory(wordService)
+  const glossaryJsx: JSX.Element = await glossaryFactory
+    .createGlossary(fragment.text)
+    .then((glossaryData) => {
+      return Glossary({ data: glossaryData })
+    })
+  const glossaryHtml: JQuery = $(
+    renderToString(wrapWithMemoryRouter(glossaryJsx))
+  )
+
+  const glossary: Paragraph | false =
+    glossaryHtml.children().length > 1 ? getGlossary(glossaryHtml) : false
+  const footNotes: Paragraph[] = getFootNotes(notesHtml)
+  const tableWithFootnotes: any = getMainTableWithFootnotes(
+    tableHtml,
+    footNotes
+  )
+
+  const headline: Paragraph = getHeadline(fragment)
+
+  const doc: Document = generateWordDocument(
+    tableWithFootnotes.table,
+    tableWithFootnotes.footNotes,
+    glossary,
+    headline,
+    records,
+    fragment
+  )
+
+  return doc
+}
+
+function wrapWithMemoryRouter(component: JSX.Element): ReactElement {
+  return <MemoryRouter>{component}</MemoryRouter>
+}
+
+function getMainTableWithFootnotes(
+  table: JQuery,
+  footNotesLines: Paragraph[]
+): any {
+  table.hide()
+  $('body').append(table)
```

Comment:
jQuery should not run wild in a React application. React is unaware of the changes and can get confused. Can you make a `div` inside the component only make jQuery modifications there?

See: https://reactjs.org/docs/integrating-with-other-libraries.html

Reply:
now jquery only runs in "#jQueryContainer" which is passed as a jQuery-Object to WordExport